types/react/v15/index.d.ts

```diff
@@ -342,13 +342,17 @@ declare namespace React {
     // Class Interfaces
     // ----------------------------------------------------------------------
 
+    type NonNullable<T> = T & {};
+    type GetComponentProps<C extends React.ComponentType<any>> = NonNullable<C['_props']>;
+
     type SFC<P = {}> = StatelessComponent<P>;
     interface StatelessComponent<P = {}> {
         (props: P & { children?: ReactNode }, context?: any): ReactElement<any> | null;
         propTypes?: ValidationMap<P>;
         contextTypes?: ValidationMap<any>;
         defaultProps?: Partial<P>;
         displayName?: string;
+        _props?: P;
```

Comment:
You're asserting a property that doesn't exist, which kinda defeats the purpose of type safety :(

Reply:
This idea here is to use a "phantom type". It will never be implemented at runtime, but we need it to statically access the props type.

Do you have any ideas how else we can do this, or how to make it more obvious what this is for?